matching-web-bundle/src/main/java/com/graphhopper/matching/http/MapMatchingResource.java

```diff
@@ -122,7 +122,8 @@ public Response match(
         } else {
             Translation tr = trMap.getWithFallBack(Helper.getLocale(localeStr));
             DouglasPeucker peucker = new DouglasPeucker().setMaxDistance(minPathPrecision);
-            PathMerger pathMerger = new PathMerger().
+            // todonow
+            PathMerger pathMerger = new PathMerger(whichGraph, whichWeighting).
```

Comment:
Problem here is that the weighting (a `FastestWeighting`) is created within `MapMatching`, so we would have to *know* this here. Probably either make `MapMatching` accept *any* weighting (through `AlgoOpts` maybe?) so here we can use the same weighting we pass to `MapMatching` or add something like `MapMatchedPath#getWeighting` ?
Not sure, and honestly not entirely convinced why we should not keep `Path#getGraph()` and `Path#getWeighting()` !?

Reply:
IMO without Path#getWeighting and Path#getGraph this sounds cleaner (and make MapMatching accepting any weighting if this is easy).

Reply:
For now, I'll let `MatchResult` contain graph and weighting. Specifically because PathMerger needs them. So I'll just shift in one level up (instead of `Path`, they go to `MatchResult`).

The entire structure is problematic anyway -- MatchResult already contains non-public edge-ids (of the query graph), and even puts them into the web response. This somehow happened when "extended_json" was added (or when I refactored it, not sure).

Maybe I'll inline everything into the Resource and extract a new "core" if necessary.

Reply:
Ahhhh, and I realize that only now: PathMerger only needs Weighting to calculate travel times, not weights. So it only uses it as a Speeding or Timing.

Should be separated at some point.

Reply:
Do you think `Weighting` should not even have `calcMillis` ? I don't know about that. Yes there is a difference between calculating a weight and a time, but to me it seems practical to have both in the same place. But if `PathMerger` just needs 'some thing that maps edge Ids to times` I agree that it would make sense to let it use a smaller interface (something like `IntDoubleFunction`).